### PR TITLE
Update spanish translation

### PR DESCRIPTION
**Pull request by sguinetti**
Greetings. I update the Spanish translation of this mill game to make it more consistent and easier to understand.

### CODE REVIEW

File: src/ui/flutter_app/lib/l10n/intl_es.arb [hunk 1]
```diff
@@ -308,23 +308,23 @@
...
     "@gameOver": {
         "description": "Game Over"
     },
-    "youWin": "Has ganado. ¡Enhorabuena!",
+    "youWin": "Ha ganado. ¡Enhorabuena!",
     "@youWin": {
         "description": "You win! Congratulations!"
     },
-    "challengeHarderLevel": "¿Desafío de un nivel más difícil?\nEl nuevo nivel será el nivel ",
+    "challengeHarderLevel": "¿Desafiar un nivel más difícil? Incrementará la dificultad",
```
Comment (calcitem) [inline src/ui/flutter_app/lib/l10n/intl_es.arb:323]: When a human player win a match against the AI, this prompt pops up, and the end of the sentence is linked to the number of the level you are about to upgrade to, e.g.

```
¿Desafiar un nivel más difícil? Aumentará la dificultad a 3.
```
So consider whether to add an `a`  and a space at the end? Thanks a lot!

File: src/ui/flutter_app/lib/l10n/intl_es.arb [hunk 2]
```diff
@@ -404,23 +404,23 @@
...
         "description": "Auto Change First Move"
     },
     "resignIfMostLose": "Renuncia de la IA si la mayoría pierde",
     "@resignIfMostLose": {
         "description": "AI Resign if Most Lose"
     },
-    "shufflingEnabled": "Movimiento aleatorio",
+    "shufflingEnabled": "Mover al azar",
     "@shufflingEnabled": {
         "description": "Random move"
```
Comment (calcitem) [inline src/ui/flutter_app/lib/l10n/intl_es.arb:421]: This translation is more accurate, and I will amend the English translation accordingly later to:  `Move randomly`.

File: src/ui/flutter_app/lib/l10n/intl_es.arb [hunk 3]
```diff
@@ -753,23 +753,23 @@
     "@takeBack": {
         "description": "Take back"
     },
-    "takingBack": "Retirar...",
+    "takingBack": "Retrocediendo...",
     "@takingBack": {
```
Comment (calcitem) [inline src/ui/flutter_app/lib/l10n/intl_es.arb:757]: The latest version of the English translation removes this ellipsis. Not sure if the Spanish custom is to put an apostrophe after it?

File: src/ui/flutter_app/lib/l10n/intl_es.arb [hunk 4]
```diff
@@ -753,23 +753,23 @@
...
         "description": "Taking back..."
     },
-    "waiting": "Esperar...",
+    "waiting": "Espere...",
```
Comment (calcitem) [inline src/ui/flutter_app/lib/l10n/intl_es.arb:760]: The latest version of the English translation removes this ellipsis. Not sure if the Spanish custom is to put an apostrophe after it?

File: src/ui/flutter_app/lib/l10n/intl_es.arb [hunk 5]
```diff
@@ -917,7 +917,7 @@
     "@atEnd": {
         "description": "At the end of the move list."
     },
-    "tapBackAgainToLeave": "Vuelve a tocar para salir.",
+    "tapBackAgainToLeave": "Vuelva a tocar para salir.",
```
Comment (calcitem) [inline src/ui/flutter_app/lib/l10n/intl_es.arb:920]: Can you help sync to line 154 of resources.dart if that's convenient?

**Review by calcitem — CHANGES_REQUESTED** [review]
Thanks for your hard work! Thank you so much!

BTW,  Can you please help confirm that the two commits below for fixing the English translation have been synced to the Spanish fix? Thanks a lot!

6962cd05428f4d7cb23d74a7e9e9d7b24d8ca44d 61b2b97eb12796a9d30ceffe05e47604919a9ea4

Due to time constraints, the descriptions in the English translation arb file have not yet been corrected.

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 1 above)
Comment (sguinetti) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:323]: Yes. "A" is to for this sentence.

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 2 above)
Comment (sguinetti) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:421]: Yes. Randomly is "al azar".

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 3 above)
Comment (sguinetti) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:757]: That's good for me. Ellipsis is used in spanish language.

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 4 above)
Comment (sguinetti) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:760]: Similar to previous sentence, it's fine.

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 5 above)
Comment (sguinetti) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:920]: I'm sorry, but I don't know about you mean.

Comment (calcitem) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:920]: Debido al mecanismo operativo de la aplicación Flutter, es necesario modificar sincrónicamente el código correspondiente a la línea 154 de resource.dart en este directorio de archivos para que surta efecto en el programa. Puede abrir el archivo y ver el contenido de esa línea para saber que hay un carácter La copia de la cadena está ahí.

**Comment by sguinetti** [comment]
Thank you. Now I see you are explaining the rules better, is my first time in learn mill game so this sencences needs to fix. For now, they are fine. In the future I will be able to improve the translation to explain to the Spanish speaking public about the rules.

**Comment by calcitem** [comment]
If you have time, can you help add and push Spanish text about the fastlane directory? In addition to screenshots and screen recordings, I can add later. If the App Store's app introduction text is translated, it will be more popular, thank you in advance!

**Comment by calcitem** [comment]
Last time I modified the MR of a German translator myself. Although the result code finally met expectations, the final git log graph was not the expected result at all. It became the author of the commit. I haven't figured out how it should be reasonable. Are you familiar with how to modify MR? When is the plan revised? If you are not familiar with it, I can merge this MR first, and then make a new Git commit by myself. In this way, I can compile a new apk earlier to you to view the effect.

**Comment by calcitem** [comment]
Please ignore the Github Action compilation error. Because I have compiled it locally, it is OK.

In addition, can you help confirm whether there are grammatical problems in the Spanish part of the following commit? These texts are displayed when using the Feedback function in the left side menu. Because this feedback library is an external library, these texts are not included in the arb file and need to be modified independently.

https://github.com/calcitem/feedback/commit/08cb47db566776952b51f8af674e43465f53e37d

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 1 above)
Comment (calcitem) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:323]: Done. See https://github.com/calcitem/Sanmill/commit/017becaeb125ed4dcf09338b947acf4a453f3c5f

File: src/ui/flutter_app/lib/l10n/intl_es.arb (hunk 5 above)
Comment (calcitem) [reply src/ui/flutter_app/lib/l10n/intl_es.arb:920]: Done. See https://github.com/calcitem/Sanmill/commit/017becaeb125ed4dcf09338b947acf4a453f3c5f

**Comment by calcitem** [comment]
Thank you for optimizing the Spanish translation!

After applying your MR, the latest apk can be downloaded here: 
https://github.com/calcitem/Sanmill/actions/runs/997751746

**Comment by calcitem** [comment]
A new version of the translated software has been released on F-Droid at:
https://f-droid.org/es/packages/com.calcitem.sanmill/

As a token of appreciation, you have been added to the following list:
https://github.com/calcitem/Sanmill/wiki/thanks

**Comment by sguinetti** [comment]
No worries. I thank you for your dedication to improving the way we communicate with this app to spanish-language people.

**Pull request merged**